CougSat1-ADCS/src/ADCSObjects.h

```diff
@@ -1,20 +1,24 @@
-#ifndef _SRC_PMIC_OBJECTS_H_
-#define _SRC_PMIC_OBJECTS_H_
+#ifndef ADCS_H
+#define ADCS_H
 
+#include <mbed.h>
+#include <rtos.h>
 #include "Configuration.h"
 #include "interfaces/CDH.h"
+#include "CISBoard/PinNames.h"
 
-#include <mbed.h>
-
-/*************************** Interfaces and Drivers ***************************/
-extern I2C      i2cLocal;
-extern I2CSlave i2cBus;
-
-extern CDH cdh;
-
-extern DigitalOut statusLED;
-
-extern InterruptIn interruptCtrlSync;
-extern InterruptIn interruptBusI2CIRQ;
-
-#endif /* _SRC_PMIC_OBJECTS_H_ */
\ No newline at end of file
+class ADCS
+{
+  private:
+	Thread monitor;
+	Thread cdhRead;
+	char message[4];
+	CDH cdh;
+	
+  public:
+	ADCS();
+	void monitorThread();
+	void cdhThread();
+	void initThread();
```

Comment:
These thread functions shouldn't be public.

Reply:
The last one needs to be for main to start it, but otherwise, you are right.